### ISSUE TITLE
can't remove scoped package

### ISSUE THREAD

**Issue by gitawego** [yarnpkg/yarn#1342]
**Do you want to request a _feature_ or report a _bug_?**
Bug
**What is the current behavior?**
can not remove scoped package from node_modules, ex:

``` shell
yarn remove @types/graphql
```

**If the current behavior is a bug, please provide the steps to reproduce.**
first, install a scoped package,ex:

``` shell
yarn add @types/graphql
```

then remove it

``` shell
yarn remove @types/graphql
```

the package remains in the node_modules folder

**What is the expected behavior?**
this package should be physically removed from folder node_modules

**Please mention your node.js, yarn and operating system version.**
node: v6.8.0   yarn: 0.16.1, ubuntu 16.04

**Comment by samccone** [comment]
Thanks for the bug! :)

**Comment by wyze** [comment]
This could be OS specific. I just tried against latest master on macOS and it worked fine. I can give it a shot on Ubuntu/Windows later and report back.

**Comment by caseyhoward** [comment]
It just failed for me on Mac OS X Sierra (10.12.1). It gets removed from the package.json but not the node_modules/@types directory. yarn check doesn't catch it. Removing node_modules and reinstalling works. But not having to clean my node_modules directory regularly is one of the main reasons I switched to yarn from npm.

**Comment by maybeec** [comment]
even the case for yarn 0.17.4

**Comment by maybeec** [comment]
seems to be fixed by #1861 but is not yet released
